.github/workflows/test.yml

```diff
@@ -209,15 +209,19 @@ jobs:
             packages/*/node_modules
             lib/
             packages/*/lib
-            packages/*/.git-data.json            
+            packages/*/.git-data.json
           key: ${{ runner.os }}-node-${{ matrix.node }}-${{ github.sha }}
           fail-on-cache-miss: true
 
       - name: Run the e2e test environment
         run: scripts/run_e2e_env.sh start
 
       - name: E2E tests
-        run: yarn test:e2e
+        # E2E tests are sometimes stalling until timeout is reached but we know that
+        # after ~10 minutes those should have passed already if there are no failed test cases.
```

Comment:
It's 15 minutes set below.

Reply:
Setting the timeout to 10 minutes might be a bit tight, I opted to give it some breathing room in case CI is overloaded and E2E tests take a bit longer. Have to wait for other tests anyways so having e2e tests run 15 minutes seem fine to me. Feel free to suggests a different value.

Reply:
I updated the comment